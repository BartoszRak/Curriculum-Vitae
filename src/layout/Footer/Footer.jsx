import React from 'react'
import { FormattedMessage } from 'react-intl'
import { withStyles, Typography } from '@material-ui/core/'
import PropTypes from 'prop-types'

import style from './Footer.style'

export const Footer = ({ classes }) => (
  <footer className={classes.root}>
    <Typography className={classes.text} variant="body2">
      <FormattedMessage id="layout.footer.copyright" values={{ date: new Date().getFullYear() }} />
    </Typography>
  </footer>
)

Footer.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(Footer)
