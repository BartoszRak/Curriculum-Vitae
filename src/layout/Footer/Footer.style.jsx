export default ({ spacing, palette }) => ({
  root: {
    background: palette.primary.main,
    padding: spacing.unit * 2,
  },
  text: {
    textAlign: 'center',
  }
})
