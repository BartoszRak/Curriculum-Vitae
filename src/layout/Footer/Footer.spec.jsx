import React from 'react'
import { render, cleanup } from '@testing-library/react'

import { Footer } from './Footer'

jest.unmock('@material-ui/core')

jest.mock('react-intl')

describe('component: Footer', () => {
  afterEach(cleanup)

  describe('rendering', () => {
    test('render without crash ', () => {
      const wrapper = render(<Footer />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot ', () => {
      const wrapper = render(<Footer />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
