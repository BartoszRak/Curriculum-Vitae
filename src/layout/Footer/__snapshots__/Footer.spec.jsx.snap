// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Footer should render as expected 1`] = `
<footer>
  <WithStyles(Typography)
    variant="body2"
  >
    <FormattedMessage
      id="layout.footer.copyright"
      values={
        Object {
          "date": 2019,
        }
      }
    />
  </WithStyles(Typography)>
</footer>
`;
